Userland/Shell/Shell.cpp

```diff
@@ -1358,7 +1359,7 @@ Vector<Line::CompletionSuggestion> Shell::complete_path(const String& base, cons
             struct stat program_status;
             String file_path = String::format("%s/%s", path.characters(), file.characters());
             int stat_error = stat(file_path.characters(), &program_status);
-            if (!stat_error) {
+            if (!stat_error && (executable_only == ExecutableOnly::No || access(file_path.characters(), X_OK) == 0)) {
                 if (S_ISDIR(program_status.st_mode)) {
```

Comment:
Won't this stop suggesting directories?
It's often expected to have the shell complete a directory name no matter what.

Reply:
Nah, that'll still work because directories are "executable".

Reply:
Alright then : )